olclient/openlibrary.py

```diff
@@ -514,20 +514,27 @@ def get(cls, olid=None, isbn=None, oclc=None, lccn=None, ocaid=None):
                 if not olid:
                     if any([isbn, oclc, lccn, ocaid]):
                         if isbn:
+                            _id, value = ('isbn', isbn)
                             olid = cls.get_olid_by_isbn(isbn)
                         elif oclc:
+                            _id, value = ('oclc', oclc)
                             olid = cls.get_olid_by_oclc(oclc)
                         elif ocaid:
+                            _id, value = ('ocaid', ocaid)
                             olid = cls.get_olid_by_ocaid(ocaid)
                         else:
+                            _id, value = ('lccn', lccn)
                             olid = cls.get_olid_by_lccn(lccn)
                     else:
                         raise ValueError("Must supply valid olid, isbn, oclc, ocaid, or lccn")
 
-                err = lambda e: logger.exception("Error retrieving OpenLibrary " \
-                                                 "book: %s", e)
+                if olid is None:
+                    raise Exception("No olid found for %s = %s" % (_id, value))
```

Comment:
Do we want to throw a specific type of exception here?

Reply:
Possibly, any idea what it should be? I used a generic exception because that's what the existing exception was at the end of `Edition.get()`:  `raise Exception("Unable to get Edition with olid: %s\nDetails: %s" % (olid, e))`

I'm trying to refactor the code to DRY things up and make results more consistent. 

Perhaps is should return `None` for the Edition instead of an exception? Since when called with bibkeys it is a search, and no results should be an expected outcome.

Reply:
I have refactored a bit and decided to return `None` since `Edition.get()` with bibkeys is a search which could easily return no results, so `None` seems a valid result.